### ISSUE TITLE
WebIDL for treating `XRWebGLLayer` as an `XRLayer`

### ISSUE THREAD

**Issue by asajeffrey** [immersive-web/layers#60]
The README says "`XRWebGLLayer` functions as an `XRProjectionLayer`, but they are kept as distinct types for better forwards compatibility," but this isn't reflected in the WebIDL.

Is the intention  that there should be a rype:
```webidl
typedef (XRWebGLLayer or XRProjectionLayer) XRBaseLayer;
```
and then use this in the places that a base layer is used? That is in the defn of `dictionary XRRenderStateInit` (https://www.w3.org/TR/webxr/#dictdef-xrrenderstateinit), the `baseLayer` field would become:
```webidl
  XRBaseLayer? baseLayer;
```
and in the defn of `XRRenderState` (https://www.w3.org/TR/webxr/#xrrenderstate):
```webidl
 readonly attribute XRBaseLayer? baseLayer;
```

**Comment by cabanier** [comment]
Thanks for filing the issue. I was holding off adding spec text until #24 was resolved.

**Comment by cabanier** [comment]
@asajeffrey This should be partially addressed with #70

**Comment by cabanier** [comment]
@asajeffrey this PR was merged. There's still a question if part of it should move to the WebXR spec.
